### DISCUSSION TITLE
homogeneous medium under directional emitter

### DISCUSSION THREAD

**Discussion by boringfish** [mitsuba-renderer/mitsuba3#415] [Q&A] [answered]
Hi，I am trying to test homogeneous medium under directional light.  If I use a Environment emitter, Mitsuba can show a great subsurface scattering effect. 
![result1](https://user-images.githubusercontent.com/32609638/204942019-871a0c78-55c8-42b4-b299-68f4ee28928e.png)
But when I use a directional emitter to replace environment emitter, the result is almost black.
![my_first_render](https://user-images.githubusercontent.com/32609638/204942443-119da008-c119-4d8c-937d-5b0990b86455.png)
The following is the XML file I used:
`<scene version="2.1.0">
    <path value="../../common"/>
	<default name="integrator" value="volpath"/>
	<default name="rr_depth" value="64"/>
	<default name="max_depth" value="64"/>

	<bsdf type="roughdielectric" id="object_bsdf">
        <string name="distribution" value="beckmann"/>
        <float name="alpha" value="0.01"/>
        <string name="int_ior" value="bk7"/>
        <string name="ext_ior" value="air"/>
    </bsdf>

    <medium id="medium1" type="homogeneous">
        <float name="scale" value="10"/>
        <phase type="hg">
			<float name="g" value="0.0"/>
		</phase>
        <float name="sigma_t" value="15"/>
        <rgb name="albedo" value="0.85,0.97,0.999999"/>
        <boolean name="has_spectral_extinction" value="false"/>
        <boolean name="sample_emitters" value="false"/>
    </medium>

    <shape type="obj">
        <string name="filename" value="meshes/bunny.obj"/>
		<transform name="to_world">
			<scale value="1"/>
			<rotate y="1" angle="90"/>
		</transform>
		<ref id="medium1" name="interior"/>
	<ref id="object_bsdf"/>
    </shape>

    <include filename="include/shape_testscene.xml"/>

</scene>`
And shape_testscene.xml is: 
`<scene version="3.0.0">
	<default name="spp" value="1024"/>
	<default name="integrator" value="path"/>
	<default name="rr_depth" value="5"/>
	<default name="max_depth" value="-1"/>

	<integrator type="$integrator">
		<integer name="rr_depth" value="$rr_depth"/>
		<integer name="max_depth" value="$max_depth"/>
	</integrator>

	<sensor type="perspective" id="Camera-camera">
		<string name="fov_axis" value="smaller"/>
		<float name="focus_distance" value="6.0"/>
		<float name="fov" value="60"/>
		<transform name="to_world">
			<lookat target="0, 0, 0" origin="3.69558, 3.46243, 3.25463" up="0, 1, 0"/>
		</transform>

		<sampler type="independent">
			<integer name="sample_count" value="$spp"/>
		</sampler>

		<film type="hdrfilm" id="film">
			<integer name="width" value="960"/>
			<integer name="height" value="720"/>
			<string name="pixel_format" value="rgb"/>
			<rfilter type="gaussian"/>
		</film>
	</sensor>

	<emitter type="directional">
    		<vector name="direction" value="0.0, -1.0, 0.0"/>
    		<rgb name="irradiance" value="1.0"/>
	</emitter>
</scene>`
So,  where did I go wrong？ Or is it actually the true result from directional emitter?

**Comment by Speierers** [comment]
Hi @boringfish ,

It is hard to tell whether something is wrong or if this is the actual true result.

Could you maybe try to set the `"sample_emitters"` value to `"true"` in you scene description?

Also it would be useful to check how the regular `path` integrator renders this scene (e.g. when removing the volume from the scene).

**Reply by boringfish** [reply]
Oh yes. I set `"sample_emitters"` to `"true"`, and tried not only directional emitter, but also spot light source. But Unfortunately, both of them show a similiar black result as:
![06d953204a5fd91a0700c521d0ffe81](https://user-images.githubusercontent.com/32609638/205063758-1021d807-adad-4693-bca6-cfc4e7529e21.png)
And I also tried to use diffuse BSDF instead of homogeneous medium under the same directional emitter. It can show a correct result:
![bd60b2cbc630b6ee5090cf0151fe6d7](https://user-images.githubusercontent.com/32609638/205064257-f9597480-19a8-41f7-abe0-c5dceca4f411.png)
It seems that homogeneous medium cannot fit directional emitter and spot light source well?

**Comment by Speierers** [comment]
It could be a limitation of our implementation of the volumetric path tracer @dvicini ?

**Comment by wjakob** [comment, marked as answer]
This is a limitation of any standard Monte Carlo path/light tracing method. When setting the light source to be a non-area source (point light, spot light, directional light) almost all light transport in the scene involves "SDS" paths, which those techniques sample with zero probability. To get an image, you would need some variation of this technique: https://rgl.epfl.ch/publications/Zeltner2020Specular that isn't currently implemented in baseline Mitsuba. What I would do in your case is to approximate the directional light source using an envmap that has a small subset of pixels turned on (but expect quite noisy images in that case).

**Comment by boringfish** [comment]
Thanks for all of your detailed discussions @wjakob @Speierers .  Since non-area source cannot fit homogeneous medium, I tried to use a disk- area source to approximate directional emitter, and found that it can show a great subsurface scattering effect. And I will also try Specular Manifold Sampling in the future. By the way, I think extending the current Particle tracer from the surface to volume (e.g. 
Volumetric photon mapping ) maybe a good solution to fit non-area sources. Does Mitsuba have plans to implement it in the future?

**Comment by Speierers** [comment]
Adding support for participating media (e.g. volumes) in the particle tracer isn't currently on our roadmap. In case you take a stab at this, feel free to open a PR so we can discuss the proposed changes.
